File: TSOL.Domain/Entities/Answer.cs


using System.ComponentModel.DataAnnotations.Schema;

namespace TSOL.Domain.Entities
{
    public class Answer : BaseEntity
    {
        public string Title { get; set; }
        public string Content { get; set; }

        public int QuestionId { get; set; }
        // public virtual Question Question { get; set; } 
        public bool IsRightAnswer { get; set; } = false;

        [NotMapped]
        public bool HasSelected { get; set; } = false;
    }
}

File: testonline-ui/src/app/model/answer.ts
export class Answer {

  id: number;
  index: number;
  questionId: number;
  title: string;
  isRightAnswer: boolean;
  hasSelected: boolean;

  constructor(data: any) {
    data = data || {};
    this.id = data.id;
    this.index = data.index;
    this.questionId = data.questionId;
    this.title = data.title;
    this.isRightAnswer = data.isRightAnswer;
    this.hasSelected = data.hasSelected;
}
}
